server/server.go

```diff
@@ -137,14 +139,14 @@ type Server struct {
 }
 
 // NewServer constructs a server from the provided config.
-func NewServer(ctx context.Context, c Config) (*Server, error) {
+func NewServer(ctx context.Context, c Config, prometheusRegistry prometheus.Gatherer) (*Server, error) {
```

Comment:
nit: prometheusRegistry should be part of Config.

Reply:
I wasn't sure about this one, and didn't look into config too much, I felt it was just the unmarshaled config, but I'll gladly add it.